Funcky.Async/Extensions/AsyncEnumerableExtensions/Shuffle.cs

```diff
@@ -1,17 +1,32 @@
-using static Funcky.Internal.Mixer;
+using Funcky.Internal;
 
 namespace Funcky.Async.Extensions
 {
     public static partial class AsyncEnumerableExtensions
     {
         /// <summary>
-        /// Returns the given sequence in random Order in O(n).
+        /// Returns the given sequence eagerly in random Order in O(n).
         /// </summary>
         /// <typeparam name="TSource">The type of the elements in the enumerable.</typeparam>
         /// <remarks>This method is implemented by using deferred execution. The immediate return value is an object that stores all the information that is required to perform the action. The query represented by this method is not executed until the object is enumerated either by calling its GetEnumerator method directly or by using foreach.</remarks>
         [Pure]
-        public static async ValueTask<IEnumerable<TSource>> Shuffle<TSource>(this IAsyncEnumerable<TSource> source, CancellationToken cancellationToken = default)
+        public static async ValueTask<IReadOnlyList<TSource>> Shuffle<TSource>(this IAsyncEnumerable<TSource> source)
             where TSource : notnull
-            => ToRandomEnumerable(await source.ToListAsync(cancellationToken).ConfigureAwait(false), new Random());
+            => (await source.ToListAsync().ConfigureAwait(false)).ToRandomList(new Random());
+
+        [Pure]
+        public static async ValueTask<IReadOnlyList<TSource>> Shuffle<TSource>(this IAsyncEnumerable<TSource> source, Random random)
```

Comment:
We usually have the cancellation token as an optional parameter with a default value.
Why are these overloads different?

Reply:
Two reasons:

1.) The sync version initially had the Random parameter as a `Option` parameter with a default value, when I added the async versions, now I had two paramters with a default, that was not ideal.
2.) Because I get an API warning otherwise, because of multiple overloads with default parameters.

We usually accept 2 for the CancellationToken, but because I have additional overloads from Random I decided I will make all parameters the same.

Reply:
Fixed and ignored the warning